helm/charts/6/6.0.0-GA/skywalking/templates/elasticsearch/elasticsearch-statefulset.yml

```diff
@@ -0,0 +1,123 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: {{ template "skywalking.fullname" . }}-elasticsearch
+  labels:
```

Comment:
add namespace

Reply:
The namespace can be specified externally. If you write to death, you will get an error if you want to modify it externally. If you want to get the namespace automatically, you can use the Releace object.
https://github.com/apache/skywalking-kubernetes/pull/8/commits/a00a507ccdad03d7f48bbc9e076101b6e63ba456

Reply:
According to my experience, we should add the following piece to specific `namespace`

```
metadata:
  ...
  namespace: {{ .Release.Namespace }}
```
Is this a mandatory configuration for deploying charts into a non-default namespace, or not?

Reply:
I understand what you said, but when we deploy skywalking, we may not deploy to namespace=skywaling or other specific namespaces. We may deploy to dev, test or prod environments. 

If we write the namespace configuration, we deploy to different environments, you need to modify the charts every time. I suggest using the command line to specify the namespace. 

` hellm install skywaling --namespace=xxx`

However, if the namespace is difined in the charts, the command line will return an error.

Reply:
{{ .Release.Namespace }} can get input from the command line

Reply:
not mandatory, kubernetes will automatically populate the namespace configuration.{{ .Release.Namespace }}  also can get this.

Reply:
> Except for namespace, other Changes requested have been modified.
> 
> About namespace ,I have two ways.
> 
> * now charts
> * add the namespace configuration to each file and maintain this configuration with values.yaml
>   `helm install skywalking --set=""`
>   I recommend using the first method, I refer to helm/charts repo.
> 
> Help review again , thanks

You are right. Most of the charts in helm repo don't contain `namespace` in their artifacts. I used to set it but it seems not the best practice.